test/test.js

```diff
@@ -0,0 +1,120 @@
+const { _testBuild } = require('../src/cli/build');
+const { _testGenerate } = require('../src/cli/generate');
+const path = require('path');
+const fs = require('fs/promises');
+const mem = require('mem');
+const { getBuildConfig } = require('../src/config/build');
+const { getBabelConfig } = require('../src/config/babel');
+const { getProjectConfig } = require('../src/config/project');
+
+const argv = {
+  _: ['project'],
+  dry: false,
+  d: false,
+  force: false,
+  f: false,
+  help: false,
+  h: false,
+  quiet: false,
+  q: false,
+  announce: false,
+  a: false,
+  '--': []
+};
+
+let oldCwd = process.cwd;
+let oldEnv = process.env.NODE_ENV;
+/**
+ * Mocks the working directory. Same as `cd newCwd`
+ */
+function mockCwd(newCwd) {
+  process.cwd = function () {
+    return newCwd;
+  };
+}
+
+/**
+ * same as rm -rf, supressing errors if the file doesn't exist
+ */
+async function rmRecursive(rootPath) {
+  try {
+    await fs.rm(rootPath, { recursive: true });
+  } catch (e) {
+    if (!e.message.includes('ENOENT')) {
+      throw e;
+    }
+  }
+}
+
+const examplesRoot = path.resolve(__dirname, './example-projects/');
+
+// clean and create working directory
+beforeAll(async () => {
+  await rmRecursive(examplesRoot);
+  await fs.mkdir(examplesRoot);
+});
+
+// Reset mocks
+afterAll(async () => {
+  process.cwd = oldCwd;
+  process.env.NODE_ENV = oldEnv;
+  await rmRecursive(examplesRoot);
+});
+['basic', 'react', 'preact', 'svelte'].forEach(template => {
+  describe(`${template} project`, () => {
+    [(true, false)].forEach(hasTypescript => {
```

Comment:
This typo is so weird.

In JS `(true, false)` equates to `false`. So turns out I was never actually testing with Typescript on.

Fixing this syntax breaks the tests because Typescript uses its own module resolution, and I need to work out how to satisfy it.